docker-bake.hcl

```
@@ -297,7 +357,22 @@ target "debian" {
     DEBIAN_RELEASE = DEBIAN_RELEASE
     JAVA_VERSION   = "${javaversion(jdk)}"
   }
-  tags      = linux_tags(type, jdk, "debian")
+  tags = [
+    # If there is a tag, add versioned tag suffixed by the jdk
+      equal(ON_TAG, "true") ?
+      "${REGISTRY}/${orgrepo(type)}:${REMOTING_VERSION}-${BUILD_NUMBER}-jdk${jdk}${equal(jdk, 25) ? "-preview" : ""}" :
+      "",
+    # If there is a tag and if the jdk is the default one, add versioned short tag
+      equal(ON_TAG, "true") ? (is_default_jdk(jdk) ? "${REGISTRY}/${orgrepo(type)}:${REMOTING_VERSION}-${BUILD_NUMBER}" : "") : "",
+    # If the jdk is the default one, add Debian and latest short tags
+      is_default_jdk(jdk) ? "${REGISTRY}/${orgrepo(type)}:bookworm" : "",
+      is_default_jdk(jdk) ? "${REGISTRY}/${orgrepo(type)}:latest" : "",
+      is_default_jdk(jdk) ? "${REGISTRY}/${orgrepo(type)}:latest-bookworm" : "",
+    "${REGISTRY}/${orgrepo(type)}:bookworm-jdk${jdk}${equal(jdk, 25) ? "-preview" : ""}",
+    "${REGISTRY}/${orgrepo(type)}:jdk${jdk}${equal(jdk, 25) ? "-preview" : ""}",
+    "${REGISTRY}/${orgrepo(type)}:latest-bookworm-jdk${jdk}${equal(jdk, 25) ? "-preview" : ""}",
+    "${REGISTRY}/${orgrepo(type)}:latest-jdk${jdk}${equal(jdk, 25) ? "-preview" : ""}",
+  ]
```

```suggestion
  tags = linux_tags(type, jdk, "debian")
```
todo: keep the function introduced in #944